### PR TITLE
feat: integrate public endpoint handling

### PR DESCRIPTION
**Pull request by vincentdelmundo**
# Description
Describe the changes in this pull request.
Explain your rationale of technical decisions you made (unless discussed before).

## Links
Add links to github/jira issues, design documents and other relevant resources (e.g. previous discussions, platform/tool documentation etc.)

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md)
- [x] I have not committed any secrets (passwords, tokens, internal urls etc.) or sensitive information
- [x] I wrote/updated tests for new/changed code
- [x] I checked that all possible internal and platform/library exceptions are gracefully handled and **will not** crash the host app
- [x] I checked for open & known issues (especially crashes) for any newly added platform APIs or libraries
- [x] I tested non-trivial changes on a real device
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

**Review by leila-gu — APPROVED** [review]
LGTM

File: signatureverifier/src/test/kotlin/io/github/rakutentech/signatureverifier/api/PublicKeyFetcherSpec.kt [hunk 1]
```diff
@@ -98,43 +115,43 @@ class PublicKeyFetcherErrorSpec : PublicKeyFetcherSpec() {
         fetcher.fetch("test_key_id")
     }
 
-    @Test(expected = Exception::class)
-    fun `should throw when the 'id' key is missing in response`() {
+    @Test
+    fun `should return valid key even if 'id' key is missing in response`() {
```
Comment (imaginaris) [inline signatureverifier/src/test/kotlin/io/github/rakutentech/signatureverifier/api/PublicKeyFetcherSpec.kt:119]: Is `id` marked as optional in the endpoint specification?

**Review by imaginaris — APPROVED** [review]
Looks ok to me

File: signatureverifier/src/test/kotlin/io/github/rakutentech/signatureverifier/api/PublicKeyFetcherSpec.kt (hunk 1 above)
Comment (vincentdelmundo) [reply signatureverifier/src/test/kotlin/io/github/rakutentech/signatureverifier/api/PublicKeyFetcherSpec.kt:119]: Will change this on a separate PR.

**Pull request merged**